### PR TITLE
Add in-app screenshot preventing

### PR DESCRIPTION
**Pull request by isamnezia**

### CODE REVIEW

File: client/platforms/ios/ScreenProtection.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+//#if os(iOS)
+    view.layer.superlayer?.addSublayer(secureTextField.layer)
+    secureTextField.layer.sublayers?.last?.addSublayer(view.layer)
+
+    secureTextField.topAnchor.constraint(equalTo: view.topAnchor, constant: 0).isActive = true
+    secureTextField.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 0).isActive = true
+    secureTextField.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 0).isActive = true
+    secureTextField.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: 0).isActive = true
+//#else
```
Comment (pokamest) [inline client/platforms/ios/ScreenProtection.swift]: Please cleanup

File: client/ui/qml/Pages2/PageHome.qml [hunk 2]
```diff
@@ -124,6 +124,9 @@ PageType {
                 text: ServersModel.defaultServerDescriptionCollapsed
             }
 
+            Component.onCompleted: {
```
Comment (pokamest) [inline client/ui/qml/Pages2/PageHome.qml]: It shouldn't be called here.
Also required refactoring for the Android part in the SettingsController::SettingsController

**Review by pokamest — CHANGES_REQUESTED** [review]

File: client/amnezia_application.cpp [hunk 3]
```diff
@@ -163,6 +167,10 @@ void AmneziaApplication::init()
     m_pageController->showOnStartup();
 #endif
 
+#ifdef Q_OS_IOS
+    m_settingsController->toggleScreenshotsEnabled(m_settingsController->isScreenshotsEnabled());
```
Comment (albexk) [inline client/amnezia_application.cpp:171]: Is it possible to move this to the generic IOS initialization block above (line 121)?

**Review by albexk — APPROVED** [review]

**Review by pokamest — APPROVED** [review]

File: client/platforms/ios/QtAppDelegate.mm [hunk 4]
```diff
@@ -19,13 +18,6 @@ - (void)applicationWillResignActive:(UIApplication *)application
...
     // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
     // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
-    _screen = [UIScreen.mainScreen snapshotViewAfterScreenUpdates: false];
-    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle: UIBlurEffectStyleDark];
-    UIVisualEffectView *blurBackground = [[UIVisualEffectView alloc] initWithEffect: blurEffect];
-    [_screen addSubview: blurBackground];
-    blurBackground.frame = _screen.frame;
-    UIWindow *_window = UIApplication.sharedApplication.keyWindow;
-    [_window addSubview: _screen];
 }
```
Comment (pokamest) [inline client/platforms/ios/QtAppDelegate.mm]: Do we need a cleanup after deleting these strings?

File: client/platforms/ios/QtAppDelegate.mm [hunk 5]
```diff
@@ -44,7 +36,6 @@ - (void)applicationWillEnterForeground:(UIApplication *)application
 - (void)applicationDidBecomeActive:(UIApplication *)application
 {
     // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
-    [_screen removeFromSuperview];
```
Comment (pokamest) [inline client/platforms/ios/QtAppDelegate.mm:47]: Do we need a cleanup after deleting these strings?

**Review by pokamest — CHANGES_REQUESTED** [review]

**Review by pokamest — APPROVED** [review]

**Pull request merged**